src/commands/cancel.ts

```diff
@@ -20,6 +22,18 @@ export const cancel: Command = {
 
     services.scheduler.cancel(name, message.guild.name);
     //remove persistance once implemented
+    const jobs = await services.store.get<StorableJob[]>("jobs");
+    if (!jobs) {
+      throw new Error("Failed to get jobs from store");
+    }
+
+    const index = jobs.findIndex((job) => job.name === name && job.message.guild === guildName);
```

Comment:
I'd implement this as a .filter(). We can combine two array operations into a single pass. We don't *need* to find the job index to know whether we need to remove it. Instead, we can use the filter to remove the requested job, and then compare the length of the two arrays after to see if the job was removed or not. This is because filter returns a *new* array.

`.splice()` is actually a pretty expensive method. Because it creates a new array AND mutates the original one. So the process here is, we first iterate through the jobs array, get index, splice to mutate original array and create a new one containing the elements removed.

However, the filter simply removes the job in the first iteration phase and returns the new array containing the jobs sans the one you removed. We can then just save it without all the intermediate steps.